components/DiffView.tsx

```diff
@@ -35,6 +35,7 @@ const DiffView: React.FC<{
   type fileComments = Record<string, { lines: number[]; comments: Comment[] }>
   //every file gets unique index in format of submissionId:fileName
   const [commentsState, setCommentsState] = React.useState<fileComments>({})
+  const [isViewable, setIsViewable] = React.useState(false)
```

Comment:
There's an issue with this way. When I try to set a single file to `viewed`, all the other files get hidden as well.

<img width="1320" alt="image" src="https://user-images.githubusercontent.com/35906419/155321725-749ecf58-0cde-402f-a279-6752ace54885.png">

If seems like all `DiffView` components share this single state instead of one for each `DiffView`.

Reply:
You can try to replicate it on https://c0d3-app-qb075zyys-c0d3.vercel.app/review/js5 if the submission still exists

Reply:
I changed the state to a map of booleans which represents the viewable state for each file. Here is the example below:

<img width="461" alt="image" src="https://user-images.githubusercontent.com/23374591/155420364-ef1aa366-11c9-4b10-a74b-a834fd178c74.png">

Then I update the state accordingly based on the fileIdx for each submission. Let me know if you have any suggestions on how to make it better!

Reply:
One issue appeared, when you click the checkbox, the background color of the filename container gets reduced to only fill the least space possible which is the filename text since the `newValue` isn't there anymore.

<img width="1398" alt="image" src="https://user-images.githubusercontent.com/35906419/155421034-a1f0eebd-69aa-4b8f-831f-8c173882df31.png">

Note: changed the bg color to green to make it obvious